src/Button/Button.tsx

```diff
@@ -0,0 +1,229 @@
+  text-align: center;
+  text-transform: ${props => (props.uppercase ? 'uppercase' : 'none')};
+  opacity: ${props => (props.disabled ? 0.5 : 1)};
+`
+
+export const Button: React.FunctionComponent<IButtonProps> = ({
+  title = '',
+  titleStyle: buttonTitleStyle,
+  titleProps,
+  buttonStyle,
+  loading = false,
+  loadingProps: buttonLoadingProps,
+  loadingStyle,
+  iconSource,
+  iconContainerStyle,
+  radius = 0,
+  iconPosition = 'left',
+  iconSize = metrics.xxl,
+  iconColor,
+  disabled = false,
+  resizeMode = 'contain',
+  disabledTitleStyle,
+  disabledStyle,
+  uppercase = false,
+  activeOpacity = 0.2,
+  containerStyle,
+  onPress = () => {},
+  onLongPress = () => {},
+}) => {
+  const iconPositionStyle = {
+    top: 'column',
+    bottom: 'column-reverse',
+    left: 'row',
+    right: 'row-reverse',
+  }
+
+  return (
+    <View style={[styles.buttonContainer, {borderRadius: radius}, containerStyle]}>
+      <TouchableOpacity
+        disabled={disabled || loading}
+        activeOpacity={activeOpacity}
+        onPress={onPress}
+        onLongPress={onLongPress}>
+        <StyledButton
+          style={[buttonStyle, !!disabled && disabledStyle]}
+          iconPosition={iconPositionStyle[iconPosition]}
+          disabled={disabled}>
+          {/* Activity Indicator on loading */}
+          {!!loading && (
+            <ActivityIndicator
+              style={[styles.loading, loadingStyle]}
+              color={buttonLoadingProps?.color || 'black'}
+              size={buttonLoadingProps?.size || 'small'}
+              {...buttonLoadingProps}
+            />
+          )}
+          {/* Button Icon, hide Icon while loading */}
+          {!loading && iconSource && (
+            <Icon
+              source={iconSource}
+              size={iconSize}
+              color={iconColor}
+              style={[styles.icon, iconContainerStyle]}
+              resizeMode={resizeMode}
+            />
+          )}
+          {/* Title for Button, hide while loading */}
+          {!loading ? (
+            typeof title === 'string' ? (
+              <StyledTitle
+                style={StyleSheet.flatten([buttonTitleStyle, !!disabled && disabledTitleStyle])}
+                disabled={disabled}
+                uppercase={uppercase}
+                as={Text}
+                {...titleProps}>
+                {title}
+              </StyledTitle>
+            ) : (
+              title
+            )
+          ) : null}
+        </StyledButton>
+      </TouchableOpacity>
+    </View>
+  )
+}
+
+const styles = StyleSheet.create({
+  buttonContainer: {
+    overflow: 'hidden',
+  },
+  loading: {
+    marginVertical: 2,
+  },
+  icon: {
+    margin: metrics.xxs,
+  },
+  button: {
+    alignItems: 'center',
+    justifyContent: 'center',
+    paddingVertical: metrics.xs,
+    paddingHorizontal: metrics.xs,
+    backgroundColor: colors.primary,
+  },
+  disableStyle: {
+    opacity: 0.5,
+  },
+  title: {
+    fontSize: 16,
```

Comment:
Put this value to a `fontSizes` constants

Reply:
done